java/memory/src/main/java/io/netty/buffer/ArrowBuf.java

```diff
@@ -494,30 +495,27 @@ public ArrowBuf retain() {
   }
 
   @Override
-  public long getLong(int index) {
-    chk(index, 8);
-    final long v = PlatformDependent.getLong(addr(index));
-    return v;
-  }
-
-  @Override
-  public float getFloat(int index) {
-    return Float.intBitsToFloat(getInt(index));
+  public ByteBuf touch() {
+    return this;
   }
 
   @Override
-  public double getDouble(int index) {
-    return Double.longBitsToDouble(getLong(index));
+  public ByteBuf touch(Object hint) {
+    return this;
   }
 
   @Override
-  public char getChar(int index) {
-    return (char) getShort(index);
+  public long getLong(int index) {
+    chk(index, 8);
+    final long v = PlatformDependent.getLong(addr(index));
```

Comment:
@zsxwing you might want to refrain from using PlatformDependent as it is in an internal package..

Reply:
The existing codes are using it. I think whether PlatformDependent should be kept or reinvented should be in a separate ticket.